fix: flaky test_concurrent_activate_deactivate in CI

— *Claude Code*

## Problem

`tests/test_quiet.py::test_concurrent_activate_deactivate` is flaky in CI.
The test spawns concurrent threads that call `activate()` / `deactivate()`,
which read and write `config.toml`. In CI, `config.toml` doesn't exist at the
expected path, causing a `FileNotFoundError` that surfaces as a test failure.

**Latest occurrence:** main CI run 23282200352 (2026-03-18), triggered by #448
merge. The test passes locally and passed on the PR branch — only fails on
the post-merge main run.

```
FAILED tests/test_quiet.py::test_concurrent_activate_deactivate
AssertionError: assert not [FileNotFoundError('config.toml not found at /home/runner/work/e-note-ion/e-note-ion/config.toml')]
```

## Expected fix

The test should mock or provide a temporary `config.toml` so it doesn't depend
on the file existing at the working directory path. Other `test_quiet.py` tests
likely handle this correctly — align this one with whatever pattern they use.

--- a/tests/test_quiet.py
+++ b/tests/test_quiet.py
@@ -131,18 +131,18 @@ def test_concurrent_activate_deactivate() -> None:
 
   def _toggle(n: int) -> None:
     try:
-      with patch('quiet._config_mod.write_config_section'):
-        for _ in range(n):
-          _mod.activate()
-          _mod.deactivate()
+      for _ in range(n):
+        _mod.activate()
+        _mod.deactivate()
     except Exception as e:  # noqa: BLE001
       errors.append(e)
 
-  threads = [threading.Thread(target=_toggle, args=(50,)) for _ in range(4)]
-  for t in threads:
-    t.start()
-  for t in threads:
-    t.join()
+  with patch('quiet._config_mod.write_config_section'):
+    threads = [threading.Thread(target=_toggle, args=(50,)) for _ in range(4)]
+    for t in threads:
+      t.start()
+    for t in threads:
+      t.join()
   assert not errors
   # Final state should be inactive (all threads did activate+deactivate pairs)
   assert not _mod.is_active()

